pkg/policyconf/policyconf.go

```diff
@@ -15,11 +15,16 @@ import (
 
 type Mode uint8
 
+func (m Mode) SupportEnforcement() bool {
+	return m != MonitorOnlyMode
+}
+
 const (
 	InvalidMode Mode = Mode(^uint8(0))
 	// NB: values below should match the ones in bpf/lib/policy_conf.h
-	EnforceMode Mode = 0
-	MonitorMode Mode = 1
+	EnforceMode     Mode = 0
+	MonitorMode     Mode = 1
+	MonitorOnlyMode Mode = 2 // monitor and cannot be updated to enforce. Not visible to bpf.
```

Comment:
Can you please add some information in the commit message for ` new(pkg/policyconf): introduce MonitorOnlyMode. `  on what is MonitorOnly mode and what is the motivation? I know it seems obvious now, but providing context is helpful both for people without the context we have now and also for our future selves.

Similarly for the other commits. Note that "why" we are doing something is equally (if not more) important that what. Usually, the what can be figured out by looking at the code, but the why is not so easy.

Reply:
I force-pushed the reworded commits; feel free to tell me if there's anything else that you feel like should be improved :)